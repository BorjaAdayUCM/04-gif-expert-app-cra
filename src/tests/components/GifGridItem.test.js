import React from 'react'
import {shallow} from 'enzyme';

import GifGridItem from '../../components/GifGridItem.js';

describe('Pruebas en <GifGridItem />', () => {

    const titulo = 'Un título';
    const url = 'https://imagen.jpg';
    const wrapper = shallow(<GifGridItem title = {titulo} url = {url}/>);
    
    test('Se debe mostrar <GifGridItem /> correctamente.', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un párrafo con el título.', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(titulo);
    });

    test('Debe tener una imagen con la url y alt proporcionadas.', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(titulo);
    })

    test('Debe tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    })
    

})