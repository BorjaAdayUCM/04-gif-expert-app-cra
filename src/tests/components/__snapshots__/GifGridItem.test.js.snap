// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem.js Se debe mostrar <GifGridItem /> correctamente. 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://imagen.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
